pkg/providers/tinkerbell/hardware/config.go

```diff
@@ -86,6 +89,11 @@ func (hc *HardwareConfig) ValidateHardware(skipPowerActions bool, tinkHardwareMa
 			return fmt.Errorf("hardware: %s ID is required", hw.Name)
 		}
 
+		if _, found := hardwareIdsDb[hw.Spec.ID]; found {
+			return fmt.Errorf("duplicate hardware id: %v", hw.Spec.ID)
+		}
+		hardwareIdsDb[hw.Spec.ID] = struct{}{}
```

Comment:
Are you planning to do anything with this hardwareIdsDb later? I won't have thought of sticking an empty struct, possibly would have gone for bool, more intuitive IMO. I miss having c++'s unordered_set here, where I could just store the keys and not have to worry about values.